packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -214,12 +214,231 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 	}
 
 	public async getSnapshot(snapshotFetchOptions?: ISnapshotFetchOptions): Promise<ISnapshot> {
-		assert(this.hostPolicy.supportGetSnapshotApi !== true, "api should not be called yet");
-		// This is just temporary as this api is not yet enabled in service policies.
-		return this.fetchSnapshot(
-			this.hostPolicy.snapshotOptions,
-			snapshotFetchOptions?.scenarioName,
+		// Fetch full snapshot in case group is not specified in request.
+		let fetchFullSnapshotContents = true;
+		if (
+			snapshotFetchOptions?.loadingGroupIds !== undefined &&
+			snapshotFetchOptions.loadingGroupIds.length > 0
+		) {
+			fetchFullSnapshotContents = false;
+		}
+		// Don't consult cache if request is not for full snapshot.
+		const { snapshot } = await this.fetchSnapshot(
+			{
+				...snapshotFetchOptions,
+				fetchSource: fetchFullSnapshotContents
+					? snapshotFetchOptions?.fetchSource
+					: FetchSource.noCache,
+				loadingGroupIds: fetchFullSnapshotContents
+					? []
+					: snapshotFetchOptions?.loadingGroupIds,
+			},
+			fetchFullSnapshotContents,
```

Comment:
So after this conversion, we have
- [] - full snapshot (all groups, slow)
- undefined - default group

Why do we need to pass fetchFullSnapshotContents, if that could be calculated from above?

Reply:
Fixed the 2 variables. 
Now, [] just means ungrouped data, ["id"] means data with group "id". 
Groupid as undefined just means legacy flow.